Backpack/Classes/Panel/BPKPanel.m

```diff
@@ -0,0 +1,96 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKPanel.h"
+
+#import <Backpack/Spacing.h>
+#import <Backpack/Color.h>
+#import <Backpack/Radii.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKPanel()
+@property (assign) UIEdgeInsets originalLayoutMargins;
+
+- (void)setup;
+@end
+
+
+@implementation BPKPanel
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    return [self initWithPadded:YES];
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+
+    if (self) {
+        [self setup];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithPadded:(BOOL)padded {
+    self = [super initWithFrame:CGRectZero];
+
+    if (self) {
+        [self setup];
+        self.padded = padded;
+    }
+
+    return self;
+}
+
+- (void)setPadded:(BOOL)padded {
+    if (padded) {
+        self.layoutMargins = self.originalLayoutMargins;
+    } else {
+        self.layoutMargins = UIEdgeInsetsZero;
+    }
+    
+    _padded = padded;
+}
+
+- (void)addSubview:(UIView *)view {
+    NSAssert(self.subviews.count == 0, @"BPKPanel can only have a single subview");
```

Comment:
why do we have this restriction?

Reply:
Because otherwise managing the complexities of the layout become unwieldy. A consumer can add a single view and then add as many more as they want. Initially I was working with a `contentView` approach instead, but AFAIK you can't make that work correctly when the view is used from IB